app/questionnaire/rules.py

```diff
@@ -170,14 +172,14 @@ def evaluate_when_rules(when_rules, metadata, answer_store, group_instance):
     :param metadata: metadata for evaluating rules with metadata conditions
     :param answer_store: store of answers to evaluate
     :param group_instance: when evaluating a when rule for a repeating group
+           this holds the group instance of the repeating group
     :return: True if the when condition has been met otherwise False
     """
     for when_rule in when_rules:
         if 'id' in when_rule:
-            when_specified_group_instance = when_rule.get('group_instance')
-            instance = when_specified_group_instance or group_instance
-
-            value = get_answer_store_value(when_rule['id'], answer_store, instance)
+            if not _answer_is_in_repeating_group(when_rule['id']):
+                group_instance = 0
```

Comment:
@AndrewTorrance Is it worth just adding an initial and to check that group instance isn't already 0? Just reduces the amount of time that the _answer_is_in_repeating_group(...) need to be evaluated. It's not a lot of overhead so just a thought.

Reply:
Fair comment , doing